deploy/addon/postgresql/pgpool.md

```diff
@@ -0,0 +1,349 @@
+---
+title: Pgpool-II for PostgresSQL add-ons
+position: 3
+shortdesc: How to configure and use Pgpool-II for PostgresSQL add-ons
+tags:
+- addons
+keywords:
+- balancing
+- connection
+- limit
+- load
+- pgpool
+- pool
+- postgres
+- proxy
+- sql
+---
+
+Pgpool-II is a tool available on applications instances to help you manage your connection pool and load balancing of your PostgreSQL add-ons.
+
+{{< alert "warning" "Warning:" >}}
+Pgpool-II is not available on Docker instances. If you want to use Pgpool-II, you can add it and configure it in your Dockerfile.
+{{< /alert >}}
+
+## What is Pgpool-II?
+
+{{< readfile "/content/partials/pgpool.md" >}}
+
+## Why use Pgpool-II?
+
+**Connection pooling**: Pgpool-II maintains established connections to the PostgreSQL servers, and reuses them whenever a new connection with the same properties (i.e. user name, database, protocol version, and other connection parameters if any) comes in. It reduces the connection overhead, and improves system's overall throughput.
+
+**Load balancing**: If a database is replicated (because running in either replication mode or native replication mode), performing a SELECT query on any server will return the same result. Pgpool-II takes advantage of the replication feature in order to reduce the load on each PostgreSQL server. It does that by distributing SELECT queries among available servers, improving the system's overall throughput. In an ideal scenario, read performance could improve proportionally to the number of PostgreSQL servers. Load balancing works best in a scenario where there are a lot of users executing many read-only queries at the same time.
+
+**Automated fail over**: If one of the database servers goes down or becomes unreachable, Pgpool-II will detach it and will continue operations by using the rest of database servers. There are some sophisticated features that help the automated failover including timeouts and retries. 
+
+**Replication**: Pgpool-II can manage multiple PostgreSQL servers. Activating the replication feature makes it possible to create a real time backup on 2 or more PostgreSQL clusters, so that the service can continue without interruption if one of those clusters fails. Pgpool-II has built-in replication (native replication). However user can use external replication features including streaming replication of PostgreSQL. 
+
+**Limiting Exceeding Connections**: There is a limit on the maximum number of concurrent connections with PostgreSQL, and new connections are rejected when this number is reached. Raising this maximum number of connections, however, increases resource consumption and has a negative impact on overall system performance. Pgpool-II also has a limit on the maximum number of connections, but extra connections will be queued instead of returning an error immediately. However, you can configure to return an error when the connection limit is exceeded (4.1 or later).  
```

Comment:
To sum up my ideas:

```suggestion
**Limiting Exceeding Connections**: PostgreSQL puts a limit on the amount of concurrent connections, so that new connections are rejected when this number is reached. Raising this maximum number of connections, however, increases resource consumption and has a negative impact on overall system performance. Pgpool-II also limits the number of concurrent connections, but extra connections will be queued instead of returning an error immediately. Or you can configure it to return an error when the connection limit is exceeded (4.1 or later).  
```